x/leveragelp/keeper/position.go

```diff
@@ -387,27 +388,42 @@ func (k Keeper) GetPositionsForAddress(ctx sdk.Context, positionAddress sdk.Addr
 
 func (k Keeper) GetPositionHealth(ctx sdk.Context, position types.Position) (sdk.Dec, error) {
 	debt := k.stableKeeper.GetDebt(ctx, position.GetPositionAddress())
-	xl := debt.Borrowed.Add(debt.InterestStacked).Sub(debt.InterestPaid)
+	debtAmount := debt.Borrowed.Add(debt.InterestStacked).Sub(debt.InterestPaid)
 
-	if xl.IsZero() {
+	baseCurrency, found := k.assetProfileKeeper.GetUsdcDenom(ctx)
+	if !found {
+		return sdk.Dec{}, errorsmod.Wrapf(assetprofiletypes.ErrAssetProfileNotFound, "asset %s not found", ptypes.BaseCurrency)
+	}
+	baseCurrencyPrice := k.oracleKeeper.GetAssetPriceFromDenom(ctx, baseCurrency)
+
+	debtValue := debtAmount.ToLegacyDec().Mul(baseCurrencyPrice)
+	if debtValue.IsZero() {
 		return sdk.ZeroDec(), nil
 	}
 
+	ammPool, err := k.GetAmmPool(ctx, position.GetAmmPoolId())
+	if err != nil {
+		return sdk.Dec{}, nil
+	}
+
+	tvl, err := ammPool.TVL(ctx, k.oracleKeeper)
+	if err != nil {
+		return sdk.Dec{}, nil
+	}
+
+	leveragedLpAmount := sdk.ZeroDec()
 	commitments := k.commKeeper.GetCommitments(ctx, position.GetPositionAddress().String())
-	positionVal := sdk.ZeroDec()
-	depositDenom := k.stableKeeper.GetDepositDenom(ctx)
+
 	for _, commitment := range commitments.CommittedTokens {
-		cacheCtx, _ := ctx.CacheContext()
-		cacheCtx = cacheCtx.WithBlockTime(cacheCtx.BlockTime().Add(time.Hour))
-		_, exitCoinsAfterExitFee, err := k.amm.ExitPool(cacheCtx, position.GetPositionAddress(), position.AmmPoolId, commitment.Amount, sdk.Coins{}, depositDenom)
-		if err != nil {
-			return sdk.ZeroDec(), err
-		}
-		positionVal = positionVal.Add(sdk.NewDecFromInt(exitCoinsAfterExitFee.AmountOf(depositDenom)))
+		leveragedLpAmount = leveragedLpAmount.Add(commitment.Amount.ToLegacyDec())
 	}
 
-	lr := positionVal.Quo(sdk.NewDecFromBigInt(xl.BigInt()))
-	return lr, nil
+	exitValue := leveragedLpAmount.Mul(tvl).Quo(ammPool.TotalShares.Amount.ToLegacyDec())
+	exitValueAfterFee := exitValue.Mul(sdk.OneDec().Sub(ammPool.PoolParams.ExitFee))
```

Comment:
I think this is wrong. when we use ExitPool, asset weights and ratios are also considered which is important when exiting the pool with getting one token(USDC in this case).
for example: If pool has 90ELYS and 10USDC and other case 50ELYS 50USDC
The result will be different in both cases.
But if we use updated formula, that will yield same result

Reply:
Please correct me if I am wrong